contracts/GnosisSafe.sol

```diff
@@ -210,7 +212,7 @@ contract GnosisSafe
             // If v is 0 then it is a contract signature
             if (v == 0) {
                 // When handling contract signatures the address of the contract is encoded into r
-                currentOwner = address(uint256(r));
+                currentOwner = address(uint160(uint256(r)));
```

Comment:
Why `uint256` -> `uint160`?

Reply:
because solidity only allows casting between same types (uintX or bytesX) or same width and byte ordering (e.g. address and uint160) .... so it is not possible to directly go from uint256 to address (to avoid potentially losing some bytes)